turk-one/app/Views-Controllers/Admin/admin-datatable.js

```diff
@@ -0,0 +1,228 @@
+									{"reviewed": {$exists: false}},
+									{"submitted": {$exists: false}},
+									{"quit": {$exists: false}}
+								]
+							});
+							break;
+						case "quit":
+							conditionalsToAllow.push({"quit": true});
+							break;
+						default:
+							break;
+					}
+				});
+
+				var filters = createMongoSelectorFromConditionals(conditionalsToAllow);
+				return WorkerTickets.find(filters, {sort: {visited: -1}});
+			}
+		}
+	});
+
+	Template.workerDataTableRow.helpers({
+		numberOfSolutionsComplexEnough: function () {
+			return Solutions.find({workerId: this._id, complexity: {$gt: 0}}).count();
+		},
+		numberOfSolutionsAccepted: function () {
+			return Solutions.find({workerId: this._id, status: "accepted"}).count();
+		},
+		numberOfSolutionsRejected: function () {
+			return Solutions.find({workerId: this._id, status: "rejected"}).count();
+		},
+		decisionPoint: function () {
+			return DecisionPoints.findOne(this.decisionPointId);
+		},
+		viewStatus: function () {
+			if (this.reviewed) {
+				return "text-success";
+			}
+			return "";
+		},
+		viewStatusLabelClass: function () {
+			if (this.quit) {
+				return "label-danger"
+			}
+			if (this.reviewed) {
+				return "label-success"
+			}
+			else if (this.submitted) {
+				return "label-primary"
+			} else {
+				return "label-default"
+			}
+		},
+		viewStatusLabelMessage: function () {
+			if (this.quit) {
+				return "quit"
+			}
+			if (this.reviewed) {
+				return "reviewed"
+			}
+			else if (this.submitted) {
+				return "submitted"
+			} else {
+				return "in-progress"
+			}
+		},
+		timeFormatted: function () {
+			moment.tz.setDefault("America/Los_Angeles");
+			return moment(this.visited.toString()).format("MM/DD/YYYY HH:mm:ss");
+		},
+		solutions: function () {
+			return Solutions.find({workerId: this._id}, {sort: {canvasNumber: 1}});
+		},
+		timeInToolFormatted: function () {
+			// Assuming workerTicket.timeInTool is the total time a worker spends anywhere on the tool
+			return moment.duration(this.timeInTool, "seconds").format("hh:mm:ss", { trim: false });
+		},
+		homeTimeFormatted: function () {
+			return moment.duration(this.homeTime, "seconds").format("hh:mm:ss", { trim: false });
+		},
+		totalSketchingTimeFormatted: function () {
+			var solutions = Solutions.find({workerId: this._id}, {sort: {canvasNumber: 1}});
+
+			var totalSolutionTime = 0;
+
+			// Assuming solution.time is the total time a worker spends using the solution sketch editor
+			solutions.forEach(function(solution) {
+				if (solution.time > 0) {
+					totalSolutionTime += solution.time;
+				}
+			});
+
+			return moment.duration(totalSolutionTime, "seconds").format("hh:mm:ss", { trim: false });
+		},
+		infoModalTimeFormatted: function () {
+			return moment.duration(this.infoModalTime, "seconds").format("hh:mm:ss", { trim: false });
+		},
+		averageSolutionTime: function () {
+			var solutions = Solutions.find({workerId: this._id}, {sort: {canvasNumber: 1}});
```

Comment:
You may need to add .fetch() to the end of this Solutions.find({}), no? .find is asynchronous and may not be returned by the time the code moves on. Adding .fetch() to the end says, to my understanding, "make this a synchronous array". If this is already working, then great :) If it sometimes misses, try .fetch() at the end!